In [2]:
from scipy import io as sio
import numpy as np

In [3]:
mat_file_str="/Users/baroc/repos/VideoDetection/validation_set_meta/validation_set_meta/validation_set.mat"
mat = sio.loadmat(mat_file_str)

In [4]:
mat

{'__globals__': [],
 '__header__': 'MATLAB 5.0 MAT-file, Platform: PCWIN64, Created on: Wed Jul 30 21:56:24 2014',
 '__version__': '1.0',
 'validation_videos': array([[ ([u'video_validation_0000001'], [u'How Do I Apply Eye Shadow to Almond-Shaped Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[261.588]], [[7845]], [[30]], [[180]], [[320]]),
         ([u'video_validation_0000002'], [u'How To Apply Eye Makeup For Asian Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[250.695]], [[7518]], [[30]], [[180]], [[320]]),
         ([u'video_validation_0000003'], [u'How To Apply Glitter Rhinestones Eye Makeup - Vegas Nay.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[383.763]], [[11510]], [[30]], [[180]], [[320]]),
         ...,
         ([u'video_validation_0001008'], [u'Wall Push-up.mpeg'], [u'WallPushups'], [[99]], [], [], [u'NO'], [[16.6119999]], [[495]], [[30]], [[180]], [[320]]),
         ([u'video_validation_0001009'], [u'Wall Pushups.mpeg'], [u'WallPushups']

In [11]:
from pprint import pprint

pprint(np.shape(mat['validation_videos']))

(1, 1010)


In [14]:
pprint(mat['validation_videos'][0][:10])

array([ ([u'video_validation_0000001'], [u'How Do I Apply Eye Shadow to Almond-Shaped Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[261.588]], [[7845]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000002'], [u'How To Apply Eye Makeup For Asian Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[250.695]], [[7518]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000003'], [u'How To Apply Glitter Rhinestones Eye Makeup - Vegas Nay.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[383.763]], [[11510]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000004'], [u'How to Apply Eye Makeup  Black Women Makeup.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[263.443]], [[7900]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000005'], [u'How to Apply Eye Makeup for Deep-Set Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[151.012]], [[4527]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000006'], [u'Ho

The "validation_videos" is a \\(1\times 1010\\) array. 

The "dtype" for each element is :
[('video_name'), ('YouTube_video_name'), ('primary_action'), ('primary_action_index'), ('secondary_actions'), ('secondary_actions_indices'), ('YouTube_video_name_includes_special_char'), ('video_duration_seconds'), ('number_of_frames'), ('frame_rate_FPS'), ('frame_height_pixels'), ('frame_width_pixels')]

The 'primary_action_index' and 'secondary_actions_indices' are indices for each label, same in the file 'class_index.txt' file. Primary action is always one, secondary actions might be multiple. 



In [68]:
meta_array = mat['validation_videos'][0]
pprint(meta_array)
print '\n\n\n\n'
pprint(meta_array[101]) # type numpy.void

array([ ([u'video_validation_0000001'], [u'How Do I Apply Eye Shadow to Almond-Shaped Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[261.588]], [[7845]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000002'], [u'How To Apply Eye Makeup For Asian Eyes.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[250.695]], [[7518]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0000003'], [u'How To Apply Glitter Rhinestones Eye Makeup - Vegas Nay.mpeg'], [u'ApplyEyeMakeup'], [[1]], [], [], [u'NO'], [[383.763]], [[11510]], [[30]], [[180]], [[320]]),
       ...,
       ([u'video_validation_0001008'], [u'Wall Push-up.mpeg'], [u'WallPushups'], [[99]], [], [], [u'NO'], [[16.6119999]], [[495]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0001009'], [u'Wall Pushups.mpeg'], [u'WallPushups'], [[99]], [], [], [u'NO'], [[38.293]], [[1146]], [[30]], [[180]], [[320]]),
       ([u'video_validation_0001010'], [u'wwwdiabeteswellnesscentercom-Wall Push Ups.mpeg'], [u

In [79]:
meta_array[107][9][0][0]

30

The data of trajectories shows that the last trajectory is 6227 if the length of video is 6228.

In [63]:
l = range(1,4500+1,100)


In [64]:
l

[1,
 101,
 201,
 301,
 401,
 501,
 601,
 701,
 801,
 901,
 1001,
 1101,
 1201,
 1301,
 1401,
 1501,
 1601,
 1701,
 1801,
 1901,
 2001,
 2101,
 2201,
 2301,
 2401,
 2501,
 2601,
 2701,
 2801,
 2901,
 3001,
 3101,
 3201,
 3301,
 3401,
 3501,
 3601,
 3701,
 3801,
 3901,
 4001,
 4101,
 4201,
 4301,
 4401,
 4501]